Sources/sentry/include/SentryApp.h

```diff
@@ -5,4 +5,28 @@
 
 #include <sentry.h>
 
-#endif
\ No newline at end of file
+#ifdef _WIN32
+#include <atomic>
+
+// The type of swift::swift_getFatalErrorMessageBuffer
+using swift_getFatalErrorMessageBuffer_t = std::atomic<const char *> *(*)();
```

Comment:
I think `std::atomic` is an implementation detail of how we read the pointer value, rather than part of the signature of the function we're importing. I think we should declare the function pointer as returning `void**` or `const char**`, and only have the `std::atomic` as a `reinterpret_cast` at the time of reading in `loadFatalErrorMessageBuffer` (so it's clear that we're using that trick to read the pointer atomically).

Reply:
`std::atomic` is part of the signature of `swift_getFatalErrorMessageBuffer`. I agree that using `void**` would work but I don't see much value hiding the type in `getFatalErrorMessageHandle` when we use `std::atomic` in `loadFatalErrorMessageBuffer`.

Reply:
oh ok, I'm surprised about this.